drivers/spi/spi_ll_stm32.c

```diff
@@ -808,6 +810,15 @@ static int spi_stm32_init(const struct device *dev)
 		return -EIO;
 	}
 
+#if defined(CONFIG_SOC_SERIES_STM32H7X)
```

Comment:
I am not convinced to this fix. It the issues would be related to SPI clock source on H7 series, It would be also reproducible on CMC rev A, which also has H7 series MCU

Reply:
The clocks _are_ configured in the device tree, and turned on by the Zephyr clock driver. But, the they never configure the SPI source to what's selected in the device tree. This is new functionality of the H7, and we've had a lot of issues with missing implementations for the H7 family.

Here is the snippet from the MCU's `spi1` node, showing how the clock is defined:
```
spi1: spi@40013000 {
    ...
    clocks = <&rcc STM32_CLOCK_BUS_APB2 0x00001000>;
};
```
Which points to the peripheral node. The clock drivers then correctly enable this clock when the SPI is started using this information. But the SPI source is never actually set. When you read the SPI clock source (for both RevA and RevB) then you get:
```
LL_RCC_GetSPIClockSource(LL_RCC_SPI123_CLKSOURCE) == LL_RCC_SPI123_CLKSOURCE_PLL1Q
```
This shows the clock source is still pointing to the PLL1, which is the default SPI1,2,3 clock source for the H7's. See the Reference Manual:
![Screenshot from 2021-12-08 15-33-34](https://user-images.githubusercontent.com/39875036/145355583-6c94cce7-818a-4e39-90ce-d83c0db5d679.png)

Reply:
@aborowski-msol @JvanDooren 
As predicted, Rev A's display was also not working as designed. In the DTS we specify a 15MHz clock. However, when the spi driver tries to match this clock to closest it can get with the expected input PCLK of  137.5 MHz and single prescaler is ~8.5MHz. So that's what it sets to. However, when I measure it with the scope the actual SCLK frequency is ~17MHz. This shows that the output clock is twice as fast as expected, because of the source clock being PLL1 instead of PCLK.

It's simply an coincidance that the display functioned at either frequency.